src/main/java/io/appform/jsonrules/utils/PreOperationUtils.java

```diff
@@ -16,111 +16,113 @@
-		switch (field) {
-		case MINUTE_OF_HOUR:
-			return dateTime.get(ChronoField.MINUTE_OF_HOUR);
-		case HOUR_OF_DAY:
-			return dateTime.get(ChronoField.HOUR_OF_DAY);
-		case DAY_OF_WEEK:
-			return dateTime.get(ChronoField.DAY_OF_WEEK);
-		case DAY_OF_MONTH:
-			return dateTime.get(ChronoField.DAY_OF_MONTH);
-		case DAY_OF_YEAR:
-			return dateTime.get(ChronoField.DAY_OF_YEAR);
-		case WEEK_OF_MONTH:
-			return dateTime.get(ChronoField.ALIGNED_WEEK_OF_MONTH);
-		case WEEK_OF_YEAR:
-			return dateTime.get(ChronoField.ALIGNED_WEEK_OF_YEAR);
-		case MONTH_OF_YEAR:
-			return dateTime.get(ChronoField.MONTH_OF_YEAR);
-		default:
-		}
-		throw new IllegalArgumentException("Operand doesnot represent a valid field");
-	}
-	
-	public static OffsetDateTime getDateTime(long epoch, String zoneOffSet) {
-		try {
-			Instant instant;
-			if (isEpochInMillis(epoch)) {
-				instant = Instant.ofEpochMilli(epoch);
-			} else {
-				instant = Instant.ofEpochSecond(epoch);
-			}
-			if (zoneOffSet != null && !zoneOffSet.trim().isEmpty()) {
-				return instant.atOffset(ZoneOffset.of(zoneOffSet));
-			}
-			return instant.atOffset(ZoneOffset.UTC);
-		} catch(Exception e) {
-			return throwInvalidDate();
-		}
-	}
-
-	private static OffsetDateTime throwInvalidDate() {
-		throw new IllegalArgumentException("Operand doesnot represent a valid date");
-	}
-
-	public static OffsetDateTime getDateTime(String dateTimeStr, String zoneOffSet) {
-		try {
-			Instant instant = Instant.parse(dateTimeStr);
-			if (zoneOffSet != null && !zoneOffSet.trim().isEmpty()) {
-				return instant.atOffset(ZoneOffset.of(zoneOffSet));
-			}
-			return instant.atOffset(ZoneOffset.UTC);
-		} catch(Exception e) {
-			throwInvalidDate();
-		}
-		throw new IllegalStateException();
-	}
-	
-	private static boolean isEpochInMillis(long epoch) {
-		// Safe check to know if epoch is in milli seconds for all dates between 1973 to 5138
-		return String.valueOf(epoch).length() >= 12;
-	}
-	
-	@SuppressWarnings("unused")
-	public static LocalDateTime getLocalDateTime(long epoch, String timeZone) {
-		try {
-			long epochSecond = Long.divideUnsigned(epoch, 1000);
-			long nanoOfSecond = Long.remainderUnsigned(epoch, 1000) * 1_000_000;
-			final ZoneOffset zoneOffSet = (timeZone == null) ? ZoneOffset.UTC : ZoneOffset.of(timeZone);
-			return LocalDateTime.ofEpochSecond(epochSecond, (int) nanoOfSecond, zoneOffSet);
-		} catch (Exception e) {
-			throwInvalidDate();
-		}
-		throw new IllegalStateException();
-	}
-
-	public static boolean isNumericRepresentation(JsonNode evaluatedNode) {
-		try {
-			Double.parseDouble(evaluatedNode.asText());
-		} catch (NumberFormatException exception) {
-			return false;
-		}
-		return true;
-	}
+    private static boolean isEpochInMillis(long epoch) {
+        // Safe check to know if epoch is in milli seconds for all dates between 1973 to
+        // 5138
+        return String.valueOf(epoch)
+                .length() >= 12;
```

Comment:
This check is problematic. Something like `Instant.ofEpochMilli(epoch)` is better. Refer to [this](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#ofEpochMilli-long-)

Reply:
This is to figure out if the epoch given is in millis or seconds. And accordingly, an `Instant` is created.